### DISCUSSION TITLE
tailwind @apply dark mode in css module is dismissed

### DISCUSSION THREAD

**Discussion by fredericrous** [vercel/next.js#22461] [Help]
**What version of Next.js are you using?**

10.0.5

**What version of Node.js are you using?**

14.3.0

**What browser are you using?**

Chrome

**What operating system are you using?**

macOS

**How are you deploying your application?**

yarn dev

**Describe the Bug**

Dark mode doesn't get applied when declared inside a css module with [@apply](https://tailwindcss.com/docs/functions-and-directives#apply) directive

**Expected Behavior**

I would expect the dark:border-white in the `steps to reproduce` example to be applied when dark mode is activated

**To Reproduce**

BurgerMenu.js

```jsx
import styles from './BurgerMenu.module.css'

export default function BurgerMenu() {
  return (
    <div className={styles.menuBtn} />
  )
}
```

BurgerMenu.module.css:

```css
.menu-btn {
  width: 40px;
  height: 40px;
  @apply border-2 border-dark dark:border-white;
}
```

it's working in tailwind playground => https://play.tailwindcss.com/RpfeDooXVs

tailwind is working just fine on the rest of the app, I used https://github.com/leerob/leerob.io as a starter app. Lee was talking about css modules in his stream so I gave it a try. It works but doesn't look compatible with dark mode.
My workaround is to do the following:

```
<div className={styles.menuBtn + " border-2  dark:border-white border-dark"}
```

but using the @apply would be sexier...

Also the button blinks when I use a css module, but that's a different issue

**Comment by hadimohamed1** [comment]
Having the same issue

**Reply by fredericrous** [reply]
@hademo could you vote on the up arrow?
![image](https://user-images.githubusercontent.com/702227/108885255-7c73b480-75ff-11eb-8ae6-81816cff596c.png)

**Reply by hadimohamed1** [reply]
done 👍

**Comment by fredericrous** [comment]
Hi @timneutkens is https://github.com/vercel/next.js/issues/22439 not a bug? in that case maybe there's some documentation work to do?
There is also a discussion about this issue on tailwind's repo => https://github.com/tailwindlabs/tailwindcss/discussions/2917

**Comment by AdaptCharm** [comment]
@frederickfogerty https://github.com/tailwindlabs/tailwindcss/issues/3258#issuecomment-770215347

**Reply by frederickfogerty** [reply]
I think you tagged the wrong person @AdaptCharm 🙃

**Reply by fredericrous** [reply]
Thank you @AdaptCharm ! The workaround of using `:global(.dark)` is simple and elegant

**Comment by cuboost** [comment]
I found out that when I saved my global.css file in Next.js, the formatter that came with VS Code would automatically format my code and add a space after `dark:`, which resulted in an error in the `@apply` section. That is why everything works fine in tailwind labs, since they don't have the VS Code format on save enabled. 

There is two ways that I found to fix that: 
1. One way is to tweak the VS Code CSS formatter settings, but the only thing you can do to make it work is to disable it completely.
2. Another way that works for me is to install another formatter, I installed Prettier and it started working...
